### ISSUE TITLE
Response from POST: empty body

### ISSUE THREAD

**Issue by WarShoe** [yhirose/cpp-httplib#1194]
I cannot get back "application/json" or "text/plain" with Server.Post() call. The http response body (seen in a browser) is always null. In the code fragment below Get() works, but Post() does not.

```
svr.Get("/get", [](const Request & req, Response &res) {
        res.set_content("{\"a\":1,\"b\":\"c\"}", "application/json");
        res.set_header("Access-Control-Allow-Origin","*");
    });

svr.Post("/post", [](const Request & req, Response &res) {
        res.set_content("{\"a\":1,\"b\":\"c\"}", "application/json");
    });
```
Post() response in browser:

```
Response { type: "opaque", url: "", redirected: false, status: 0, ok: false, statusText: "", headers: Headers, body: null, bodyUsed: false }
```

Javascript function to send/receive the post request:
```
    async function mysend () {
        const url = 'http://localhost:8080/post', data = { answer: 42 };
        console.log('fetching...');
        const response = await fetch(url, {
            method: 'POST', // *GET, POST, PUT, DELETE, etc.
            mode: 'no-cors', // no-cors, *cors, same-origin
            cache: 'no-cache', // *default, no-cache, reload, force-cache, only-if-cached
            headers: {
                'Content-Type': 'application/json'
            },
            redirect: 'follow', // manual, *follow, error
            referrerPolicy: 'no-referrer',
            body: JSON.stringify(data) // body data type must match "Content-Type" header
        });
        
        console.log('response:',response);
        const repl = await response.json();
        console.log('response data:',repl);
    }
```

Self-test result:
[==========] 226 tests from 65 test suites ran. (138626 ms total)
[  PASSED  ] 226 tests.

**Comment by PixlRainbow** [comment]
I responded to you in the closed issue #1193 but for convenience I will repeat it here:
> > `Response { type: "opaque", url: "", redirected: false, status: 0, ok: false, statusText: "", headers: Headers, body: null, bodyUsed: false }`
> 
> Hi, it appears you are attempting to make a cross-origin ajax request. For security reasons, web browsers will block access to the response of a server with a different port or domain from the current page by default. In order to disable this behavior, the browser must see "cors headers" in the server response and the request must be made with cors mode enabled

**Comment by WarShoe** [comment]
thanks! I replied in the thread. My code works now.
